be/src/exec/hash_join_components.cpp

```diff
@@ -87,8 +87,11 @@ Status HashJoinBuilder::append_chunk(RuntimeState* state, const ChunkPtr& chunk)
 }
 
 Status HashJoinBuilder::build(RuntimeState* state) {
-    SCOPED_TIMER(_hash_joiner.build_metrics().build_ht_timer);
-    TRY_CATCH_BAD_ALLOC(RETURN_IF_ERROR(_ht.build(state)));
+    // if query is already cancelled, building hash table is unnecessary, just skip it
+    if (!state->is_cancelled()) {
+        SCOPED_TIMER(_hash_joiner.build_metrics().build_ht_timer);
+        TRY_CATCH_BAD_ALLOC(RETURN_IF_ERROR(_ht.build(state)));
+    }
     _ready = true;
```

Comment:
if state->is_cancelled, _ready should be false?

Reply:
> if state->is_cancelled, _ready should be false?

_ready was introduced when implementing spill, but I didn't see it being called anywhere. It should have no impact. I'll confirm it with @stdpain later.

Reply:
> > if state->is_cancelled, _ready should be false?
> 
> _ready was introduced when implementing spill, but I didn't see it being called anywhere. It should have no impact. I'll confirm it with @stdpain later.

confirmed, it is not used for now.